lib/rubygems/ext/build_error.rb

```diff
@@ -2,5 +2,7 @@
 ##
 # Raised when there is an error while building extensions.
 
+require 'rubygems/exceptions'
```

Comment:
Why this new require?

Reply:
otherwise I got error running test, but I can take a look if that's not test problem

Reply:
Oh, I see the "require chain" now and how the error happens.

This specific require makes sense to me since a constant it defines is used right after.

What I would maybe do is moving the `rubygems/ext` require in `lib/rubygems/specification_policy.rb` to the first line inside the new `validate_extensions` method to make sure we don't load more code than necessary upfront.

Reply:
done